tests/python/gaia-ui-tests/gaiatest/tests/functional/browser/test_browser_launch_from_background.py

```diff
@@ -0,0 +1,37 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from gaiatest import GaiaTestCase
+from marionette_driver import By, Wait
+
+from gaiatest.apps.search.app import Search
+from gaiatest.apps.search.regions.browser import Browser
+
+
+class TestLaunchViaManifest(GaiaTestCase):
+    def setUp(self):
+        GaiaTestCase.setUp(self)
+        self.connect_to_local_area_network()
+
+        self.test_url = self.marionette.absolute_url('mozilla.html')
+
+    def test_launch_manifest(self):
+        search = Search(self.marionette)
+        search.launch()
+
+        self.device.touch_home_button()
+
+        search.launch()
+
+        browser = search.go_to_url(self.test_url)
+        browser.switch_to_content()
+        Wait(self.marionette).until(lambda m: m.title == 'Mozilla')
+
+        self.device.touch_home_button()
+
+        search.launch()
```

Comment:
if you add search.go_to_url after line 33, it should raise exception. Because it cannot tap the search url after the browser already surf any website. but you can patch the app.py under search folder. the code as below, then it should work..

    def go_to_url(self, url):
        # The URL bar shown is actually in the system app not in this Search app.
        # We switch back to the system app, then tap the panel, but this will only
        # work from Search app which embiggens the input bar
        self.marionette.switch_to_frame()
        if self.is_element_present(*self._url_bar_locator):
            self.marionette.find_element(*self._url_bar_locator).tap()
        else:
            self._url_bar_locator = (By.CSS_SELECTOR, '.urlbar .title')
            self._root_element = self.marionette.find_element(*self._browser_app_locator)
            self._root_element.find_element(*self._url_bar_locator).tap()
        from gaiatest.apps.homescreen.regions.search_panel import SearchPanel
        search_panel = SearchPanel(self.marionette)
        return search_panel.go_to_url(url)